proposals/NNNN-resolve-distributed-actor-protocols.md

```diff
@@ -0,0 +1,474 @@
+}
+
+// "stub" type
+distributed actor $Greeter<ActorSystem>: Greeter, DistributedStubActor
+    where DistributedActorSystem<any Codable> {
+  private init() {} // cannot initialize, can only resolve(id:using:)
+}
+
+extension Greeter where Self: DistributedActorStub {
+  // ... stub implementations for protocol requirements ...
+}
+```
+
+Default implementations for all the protocol's requirements (including non-distributed requirements) are provided by extensions utilizing the `DistributedActorStub` protocol.
+
+It is possible for a protocol type to inherit other protocols, in that case the parent protocol must either have default implementations for all its requirements, or it must also apply the `@DistributedProtocol` protocol which generates such default implementations.
+
+The default method "stub" implementations provided by the `@DistributedProtocol` simply fatal error if they were to ever be invoked. In practice, invoking those methods is not possible, because resolving a stub will always return a remote reference, and therefore calls on these methods are redirected to `DistributedActorSystem`'s `remoteCall` rather than invoking the "local" methods.  
+
+It is recommended to use `some/any Greeter` rather than `$Greeter` when passing around resolved instances of a distributed actor protocol. This way none of your code is tied to the fact of using a specific type of proxy, but rather, can accept any greeter, be it local or resolved through a proxy reference. This can come in handy when refactoring a codebase, and merging modules in such way where the greeter may actually be a local instance in some situations.
+
+### Interaction with the `DefaultDistributedActorSystem`
+
+Since the introduction of distributed actors, it is possible to declare a module wide `DefaultDistributedActorSystem` type alias, like this:
+
+```swift
+typealias DefaultDistributedActorSystem = ClusterSystem
+```
+
+This makes it easier to declare distributed actors as the `ActorSystem` type requirement is witnessed by an implicit type alias generated in every concrete distributed actor, like this:
+
+```swift
+distributed actor Worker { 
+  // synthesized:
+  // typealias ActorSystem = ClusterSystem // because 'DefaultDistributedActorSystem = ClusterSystem'
+  
+  distributed func work()
+}
+```
+
+The newly introduced ability to abstract over the `ActorSystem` in concrete distributed actors _wins_ over the synthesized typealias, causing the typealias to not be emitted:
+
+```swift
+distributed actor Worker<ActorSystem> where ActorSystem: DistributedActorSystem<any Codable> {
+  distributed func work()
+}
+```
+
+The `ActorSystem` type requirement of the `DistributedActorProtocol` in this case is witnessed by the generic parameter, and not by the "default" fallback type.
+
+This is the right behavior because this generic worked can work on _any_ distributed actor system where the `SerializationRequirement` is Codable, and not only on the `ClusterSystem`.
```

Comment:
```suggestion
This is the right behavior because this generic type works with _any_ distributed actor system where the `SerializationRequirement` is Codable, and not only on the `ClusterSystem`.
```